src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -95,73 +98,139 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
-   * ['TestSection2', null, 'TestName2', 'test2@example.com', null]
-   *
-   * 2 by 5 spreadsheetData (after)
-   * "TestSection1|Team1||test1@xample.com|test1comments\n
-   *  TestSection2||TestName2|test2@example.com|\n"
-   */
-  getUserDataRows(spreadsheetData: string[][]): string {
-    // needs to check for '' as an initial empty row with null values will be converted to e.g. "||||" after .map
-    return spreadsheetData.filter((row: string[]) => (!row.every((cell: string) => cell === null || cell === '')))
-        .map((row: string[]) => row.join('|'))
-        .map((row: string) => row.replace(/\n|\r/g, ''))
-        .join('\n');
-  }
-
   /**
    * Submits enroll data
    */
   submitEnrollData(): void {
     const newStudentsHOTInstance: Handsontable =
         this.hotRegisterer.getInstance(this.newStudentsHOT);
-    const spreadsheetData: string[][] = newStudentsHOTInstance.getData();
+    const spreadsheetData: string[][] = newStudentsHOTInstance.getData()
+        .filter((row: string[]) => (!row.every((cell: string) => cell === null || cell === '')));
 
     const hotInstanceColHeaders: string[] = (newStudentsHOTInstance.getColHeader() as string[]);
-    const dataPushToTextarea: string =
-        this.getUpdatedHeaderString(hotInstanceColHeaders);
-    const userDataRows: string = this.getUserDataRows(spreadsheetData);
 
-    this.enrollData = (userDataRows === ''
-        ? '' : dataPushToTextarea + userDataRows); // only include header string if userDataRows is not empty
-
-    const paramMap: { [key: string]: string } = {
-      courseid: this.courseid,
-      user: this.user,
+    const studentsEnrollRequest: StudentsEnrollRequest = {
+      studentEnrollRequests: [],
     };
-    this.httpRequestService.post('/course/enrollSave', paramMap, this.enrollData)
-        .subscribe((resp: EnrollResultPanelList) => {
-          this.showEnrollResults = true;
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
-          this.enrollResultPanelList = resp.enrollResultPanelList;
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessage.push({
-            message: resp.error.message,
-            color: 'danger',
-          });
+
+    // Parse the user input to be requests.
+    for (const row of spreadsheetData) {
+      const request: StudentEnrollRequest = {
+        section: row[hotInstanceColHeaders.indexOf(this.colHeaders[0])],
+        team: row[hotInstanceColHeaders.indexOf(this.colHeaders[1])],
+        name: row[hotInstanceColHeaders.indexOf(this.colHeaders[2])],
+        email: row[hotInstanceColHeaders.indexOf(this.colHeaders[3])],
+        comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])],
+      };
+      if (request.comments === null) {
```

Comment:
Will comments be `null`?

If there is a null-check for comments, how about for others?

They won't be null because of https://github.com/TEAMMATES/teammates/pull/9647/files#diff-cc2421f6f5575039136568f60626b85fR108?